piet-direct2d/src/dwrite.rs

```diff
@@ -3,24 +3,36 @@
 // TODO: get rid of this when we actually do use everything
 #![allow(unused)]
 
+use std::ffi::OsString;
 use std::fmt::{Debug, Display, Formatter};
 use std::mem::MaybeUninit;
 use std::ptr::null_mut;
+use std::sync::Arc;
 
+use winapi::shared::ntdef::LOCALE_NAME_MAX_LENGTH;
 use winapi::shared::winerror::{HRESULT, SUCCEEDED, S_OK};
 use winapi::um::dwrite::{
-    DWriteCreateFactory, IDWriteFactory, IDWriteTextFormat, IDWriteTextLayout,
-    DWRITE_FACTORY_TYPE_SHARED, DWRITE_FONT_STRETCH_NORMAL, DWRITE_FONT_STYLE_NORMAL,
-    DWRITE_FONT_WEIGHT_NORMAL, DWRITE_HIT_TEST_METRICS, DWRITE_LINE_METRICS,
-    DWRITE_TEXT_ALIGNMENT_CENTER, DWRITE_TEXT_ALIGNMENT_JUSTIFIED, DWRITE_TEXT_ALIGNMENT_LEADING,
-    DWRITE_TEXT_ALIGNMENT_TRAILING, DWRITE_TEXT_METRICS,
+    DWriteCreateFactory, IDWriteFactory, IDWriteFontCollection, IDWriteFontFamily,
+    IDWriteLocalizedStrings, IDWriteTextFormat, IDWriteTextLayout, DWRITE_FACTORY_TYPE_SHARED,
+    DWRITE_FONT_STRETCH_NORMAL, DWRITE_FONT_STYLE, DWRITE_FONT_STYLE_ITALIC,
+    DWRITE_FONT_STYLE_NORMAL, DWRITE_FONT_WEIGHT, DWRITE_FONT_WEIGHT_NORMAL,
+    DWRITE_HIT_TEST_METRICS, DWRITE_LINE_METRICS, DWRITE_TEXT_ALIGNMENT_CENTER,
+    DWRITE_TEXT_ALIGNMENT_JUSTIFIED, DWRITE_TEXT_ALIGNMENT_LEADING, DWRITE_TEXT_ALIGNMENT_TRAILING,
+    DWRITE_TEXT_METRICS, DWRITE_TEXT_RANGE,
 };
+use winapi::um::unknwnbase::IUnknown;
+use winapi::um::winnls::GetUserDefaultLocaleName;
 use winapi::Interface;
 
 use wio::com::ComPtr;
-use wio::wide::ToWide;
+use wio::wide::{FromWide, ToWide};
 
-use piet::TextAlignment;
+use piet::{FontWeight, TextAlignment};
+
+use crate::Brush;
+
+/// "en-US" as null-terminated utf16.
```

Comment:
Note: https://www.reddit.com/r/rust/comments/hse568/announce_utf16_lit100_for_all_your_utf16_literal/

Reply:
worth it?

Reply:
If there are a bunch of these, otherwise not.

Reply:
too late I did it 😞